matchbook-sdk-core/src/main/java/com/matchbook/sdk/core/ClientRest.java

```diff
@@ -0,0 +1,52 @@
+package com.matchbook.sdk.core;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import com.matchbook.sdk.core.configs.ClientConnectionManager;
+import com.matchbook.sdk.core.services.Auth;
+import com.matchbook.sdk.core.services.AuthRest;
+import com.matchbook.sdk.core.services.ClientService;
+
+public class ClientRest implements Client {
+
+    private final AtomicReference<Auth> authClient;
+
+    private ClientConnectionManager clientConnectionManager;
+
+    public ClientRest(ClientConfig clientConfig) {
+        this.clientConnectionManager = new ClientConnectionManager(clientConfig);
+
+        this.authClient = new AtomicReference<>();
+    }
+
+    public Auth getAuthClient() {
+        return newClient(authClient, AuthRest::new);
+    }
+
```

Comment:
May you please improve how this class is formatted? There are a lot of extra whitespaces. Perhaps we can let Codacy check that for ourselves.